FinSearchUnified/Tests/BusinessLogic/Models/FindologicArticleModelTest.php

```diff
@@ -0,0 +1,138 @@
+    {
+        try {
+            /** @var Article $resource */
+            $resource = $this->manager->getResource('Article');
+            $article = $resource->create($testProductConfiguration);
+
+            return $article;
+        } catch (\Exception $e) {
+            echo sprintf("Exception: %s", $e->getMessage());
+        }
+
+        return null;
+    }
+
+    public function articleSupplierProvider()
+    {
+        return [
+            'Article with normal supplier name' => [
+                [
+                    'name' => 'FindologicArticle 1',
+                    'active' => true,
+                    'tax' => 19,
+                    'supplier' => 'Findologic',
+                    'categories' => [
+                        ['id' => 3],
+                        ['id' => 5],
+                    ],
+                    'images' => [
+                        ['link' => 'https://via.placeholder.com/300/F00/fff.png'],
+                        ['link' => 'https://via.placeholder.com/300/09f/000.png'],
+                    ],
+                    'mainDetail' => [
+                        'number' => 'FINDOLOGIC1',
+                        'active' => true,
+                        'inStock' => 16,
+                        'prices' => [
+                            [
+                                'customerGroupKey' => 'EK',
+                                'price' => 99.34,
+                            ],
+                        ]
+                    ],
+                ]
+            ],
+            'Article with supplier name space' => [
+                [
+                    'name' => 'FindologicArticle 2',
+                    'active' => true,
+                    'tax' => 19,
+                    'supplier' => " ",
+                    'categories' => [
+                        ['id' => 3],
+                        ['id' => 5],
+                    ],
+                    'images' => [
+                        ['link' => 'https://via.placeholder.com/300/F00/fff.png'],
+                        ['link' => 'https://via.placeholder.com/300/09f/000.png'],
+                    ],
+                    'mainDetail' => [
+                        'number' => 'FINDOLOGIC2',
+                        'active' => true,
+                        'inStock' => 16,
+                        'prices' => [
+                            [
+                                'customerGroupKey' => 'EK',
+                                'price' => 99.34,
+                            ],
+                        ]
+                    ],
+                ]
+            ]
+        ];
+    }
+
+    /**
+     * @dataProvider articleSupplierProvider
+     *
+     * @param array $articleConfiguration The article configuration with the corresponding supplier.
+     */
+    public function testEmptySuppliersAreSkipped($articleConfiguration)
+    {
+        try {
+            $baseCategory = new Category();
```

Comment:
Check if this also creates a DB entry. If yes -> consider removing it in the tearDown method.

Reply:
I checked this. As the category is only created temporarily, its not being persistet. I debugged it and checked the database to make sure also.